app/src/main/java/com/dkanada/gramophone/ui/activities/SplashActivity.java

```diff
@@ -32,40 +33,31 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     protected void onResume() {
         super.onResume();
-        if (detectBatteryOptimization()) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && detectBatteryOptimization()) {
```

Comment:
Is this check required even with the decorators below? I've never used them before, but they seem to just shift the problem around. Seems good for documentation though so I think it's fine.

Reply:
The decorators basically say that you may not use the method below a certain sdk. So, the checks are still necessary. I just moved them to a more central space pretty much.